File: src/test/java/org/cloudfoundry/credhub/helper/FakeCurrentTimeProvider.java
package org.cloudfoundry.credhub.helper;

import java.time.Instant;
import java.time.temporal.TemporalAccessor;
import java.util.Optional;

import org.cloudfoundry.credhub.util.CurrentTimeProvider;

public class FakeCurrentTimeProvider extends CurrentTimeProvider {

  private long timeMillis;

  public void setCurrentTimeMillis(final long timeMillis) {
    this.timeMillis = timeMillis;
  }

  @Override
  public Optional<TemporalAccessor> getNow() {
    throw new UnsupportedOperationException("not yet implemented");
  }

  @Override
  public Instant getInstant() {
    throw new UnsupportedOperationException("not yet implemented");
  }

  @Override
  public long currentTimeMillis() {
    return timeMillis;
  }

  @Override
  public void sleep(final long sleepTimeInMillis) throws InterruptedException {
    timeMillis += sleepTimeInMillis;
  }
}

File: components/test-support/src/test/kotlin/org/cloudfoundry/credhub/utils/FakeCurrentTimeProvider.kt
package org.cloudfoundry.credhub.utils

import org.cloudfoundry.credhub.util.CurrentTimeProvider
import java.time.Instant
import java.time.temporal.TemporalAccessor
import java.util.Optional

class FakeCurrentTimeProvider : CurrentTimeProvider() {
    private var timeMillis: Long = 0

    override val instant: Instant
        get() = throw UnsupportedOperationException("not yet implemented")

    fun setCurrentTimeMillis(timeMillis: Long) {
        this.timeMillis = timeMillis
    }

    override fun getNow(): Optional<TemporalAccessor> = throw UnsupportedOperationException("not yet implemented")

    override fun currentTimeMillis(): Long = timeMillis

    @Throws(InterruptedException::class)
    override fun sleep(sleepTimeInMillis: Long) {
        timeMillis += sleepTimeInMillis
    }
}
